test/NVQPP/integration/sudoku_2x2-bit_names.cpp

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ -v %s -o %basename_t.x --target quantinuum --emulate && ./%basename_t.x | FileCheck %s
+
+#include <cudaq.h>
+#include <algorithm>
+#include <iostream>
+#include <unordered_set>
+
+__qpu__ void reflect_uniform(cudaq::qreg<> &qubits) {
+  h(qubits);
+  x(qubits);
+  z<cudaq::ctrl>(qubits[0], qubits[1], qubits[2], qubits[3]);
+  x(qubits);
+  h(qubits);
+}
+
+__qpu__ void oracle(cudaq::qreg<> &cs, cudaq::qubit &target) {
+  x<cudaq::ctrl>(cs[0], !cs[1], !cs[2], cs[3], target);
+  x<cudaq::ctrl>(!cs[0], cs[1], cs[2], !cs[3], target);
+}
+
+__qpu__ void grover() {
+  cudaq::qreg qubits(4);
+  cudaq::qubit ancilla;
+
+  // Initialization
+  x(ancilla);
+  h(ancilla);
+  h(qubits); // uniform initialization
+
+  oracle(qubits, ancilla);
+  reflect_uniform(qubits);
+  oracle(qubits, ancilla);
+  reflect_uniform(qubits);
+
+// TODO: Extend measurement support for submissions to IonQ,
+// see https://github.com/NVIDIA/cuda-quantum/issues/512.
+#ifndef IONQ_TARGET
+  auto groverQubits0 = mz(qubits[0]);
+  auto groverQubits1 = mz(qubits[1]);
+  auto groverQubits2 = mz(qubits[2]);
+  auto groverQubits3 = mz(qubits[3]);
+#endif
+};
+
+int main() {
+  auto result = cudaq::sample(1000, grover);
+  result.dump();
+
+  // Make sure that the get_marginal() results for the individual register names
+  // match the subset of the bits from the global register.
+  // Note that this will fail if you only compile this in library mode.
```

Comment:
might add 
```cpp
auto& platform = cudaq::get_platform();
if (platform.is_remote()) {
  ... your code ...
}
```
or something to that effect, so the executable will always work

Reply:
Please take a look at the latest push (b07ca48a) and let me know what you think.